zzzz_modular_occulus/code/modules/projectiles/guns/energy/railgun.dm

```diff
@@ -0,0 +1,176 @@
+/obj/item/gun/energy/laser/railgun
+	name = "NT \"Reductor\" rail rifle"
+	desc = "\"Moebius\" brand rail gun. This gun features a sleek and deadly design with the capability for lethal and non-lethal firing modes. A competant engineer can also overclock it using a wrench, \
+	consuming the cell within for an extra bullet."
+	icon = 'zzzz_modular_occulus/icons/obj/guns/energy/railgun.dmi'
+	icon_state = "railgun"
+	item_state = "railgun"
+	item_charge_meter = TRUE
+	fire_sound = 'zzzz_modular_occulus/sound/weapons/rail.ogg'
+	item_charge_meter = TRUE
+	w_class = ITEM_SIZE_BULKY
+	force = WEAPON_FORCE_PAINFUL
+	twohanded = TRUE
+	flags = CONDUCT
+	slot_flags = SLOT_BACK
+	origin_tech = list(TECH_COMBAT = 4, TECH_MAGNET = 6, TECH_ENGINEERING = 6)
+	matter = list(MATERIAL_PLASTEEL = 20, MATERIAL_STEEL = 8, MATERIAL_SILVER = 10)
+	charge_cost = 1250
+	gun_tags = list(GUN_PROJECTILE, GUN_LASER, GUN_ENERGY, GUN_SCOPE)
+	suitable_cell = /obj/item/cell/large
+	one_hand_penalty = 10
+	fire_delay = 14 //Slow, on par with a shotgun pump then fire
+	recoil_buildup = 12 //Big shots, big recoil.
+	damage_multiplier = 1.2
+	init_firemodes = list(
+		list(mode_name="slug", mode_desc="fires a large metal chunk at light speeds", projectile_type=/obj/item/projectile/bullet/shotgun/railgun, icon="kill"),
+		list(mode_name="non-lethal", mode_desc="fires a rubber pellet at light speed", projectile_type=/obj/item/projectile/bullet/shotgun/beanbag/railgun, icon="stun"),
```

Comment:
May want to reword this, unless this motherfucker is firing at relativistic velocities.